src/utilities/BatchTableHierarchyExtension.js

```diff
@@ -0,0 +1,122 @@
+import { parseBinArray } from './FeatureTable.js';
+
+export class BatchTableHierarchyExtension {
+
+	constructor( batchTable ) {
+
+		this.batchTable = batchTable;
+
+		const extensionHeader = batchTable.header.extensions[ '3DTILES_batch_table_hierarchy' ];
+
+		this.classes = extensionHeader.classes;
+		for ( const classDef of this.classes ) {
+
+			const instances = classDef.instances;
+			for ( const property in instances ) {
+
+				classDef.instances[ property ] = this._parseProperty( instances[ property ], classDef.length );
+
+			}
+
+		}
+
+		this.instancesLength = extensionHeader.instancesLength;
+
+		this.classIds = this._parseProperty( extensionHeader.classIds, this.instancesLength );
+
+		if ( extensionHeader.parentCounts ) {
+
+			this.parentCounts = this._parseProperty( extensionHeader.parentCounts, this.instancesLength );
+
+		} else {
+
+			this.parentCounts = new Array( this.instancesLength ).fill( 1 );
+
+		}
+
+		if ( extensionHeader.parentIds ) {
+
+			this.parentIds = this._parseProperty( extensionHeader.parentIds );
```

Comment:
The `this._parseProperty` function takes a `propertyLength` but it's not passed in here. And if `parentIds` is always an array then there's no reason to pass it in to the function since the parseProperty doesn't do anything in that case.

It looks like [this isn't always an array, though](https://github.com/CesiumGS/3d-tiles/blob/87218a7c2030890b5200cf16cb466dbbd1480a20/extensions/3DTILES_batch_table_hierarchy/schema/3DTILES_batch_table_hierarchy.schema.json#L44-L46).

Reply:
Yes indeed, I added the length of `parentIds` to `_parseProperty` parameters.